lib/components/src/blocks/ColorPalette.tsx

```diff
@@ -107,10 +105,52 @@ const List = styled.div(({ theme }) => ({
   flexDirection: 'column',
 }));
 
+type Colors = string[] | { [key: string]: string };
+
 interface ColorProps {
   title: string;
   subtitle: string;
-  colors: string[];
+  colors: Colors;
+}
+
+function renderSwatch(color: string) {
+  return (
+    <Swatch
+      key={color}
+      title={color}
+      style={{
+        backgroundColor: color,
+      }}
+    />
+  );
+}
+
+function renderSwatchLabel(color: string, colorDescription?: string) {
+  return (
+    <SwatchLabel key={color} title={color}>
+      {color}
+      {colorDescription && <small>{colorDescription}</small>}
+    </SwatchLabel>
+  );
+}
+
+function renderSwatchSpecimen(colors: Colors) {
+  if (Array.isArray(colors)) {
+    return (
+      <SwatchSpecimen>
+        <SwatchColors>{colors.map(color => renderSwatch(color))}</SwatchColors>
+        <SwatchLabels>{colors.map(color => renderSwatchLabel(color))}</SwatchLabels>
+      </SwatchSpecimen>
+    );
+  }
+  return (
+    <SwatchSpecimen>
+      <SwatchColors>{Object.values(colors).map(color => renderSwatch(color))}</SwatchColors>
```

Comment:
Thank you for your review @RuneKR but I don't get it?
I simply iterate over the object as it is. 
I mean if design tokens need to be reordered then, it must be done in the story, I guess.

Reply:
Looks like you got the gist of it. I understand that the order is arbitrary in this solution. I was wondering if the solution could/should be extended to allow the developer to specify the order of the colours. This could be achieved by using this data structure:

```
[{name: 'name of colour', value: 'hex or whatever']
```

This capability can also be added without making it a breaking change.

Reply:
Actually, it's an excellent question! 
I still use design tokens starting by `$` (because of sass variable conventions) so it does not concern me directly.
It would be great to have the point of view of someone else because I consider that it's not totally necessary. :)

Reply:
I feel like that could be enhancement made in a following PR, if desired.

Reply:
I wouldn't call the order arbitrary. (Ignoring IE)
https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/